shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-distsql/shardingsphere-shadow-distsql-parser/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/core/ShadowDistSQLStatementVisitor.java

```diff
@@ -127,15 +128,18 @@ public ASTNode visitShowShadowTableRules(final ShowShadowTableRulesContext ctx)
     }
     
     private static String getText(final ParserRuleContext ctx) {
+        if (null == ctx || ctx.isEmpty()) {
+            return null;
+        }
         return new IdentifierValue(ctx.getText()).getValue();
     }
     
     private Collection<ShadowAlgorithmSegment> visitShadowAlgorithms(final List<ShadowAlgorithmDefinitionContext> ctxs) {
         return ctxs.stream().map(this::visit).map(each -> (ShadowAlgorithmSegment) each).collect(Collectors.toList());
     }
     
-    private String createAlgorithmName(final String tableName, final AlgorithmSegment algorithmSegment) {
-        return (tableName + "_" + algorithmSegment.getName()).toLowerCase();
+    private String createAlgorithmName(final String ruleName, final String tableName, final AlgorithmSegment algorithmSegment) {
+        return (ruleName + "_" + tableName + "_" + algorithmSegment.getName()).toLowerCase();
```

Comment:
Is it better to use String.format instead of `+`?

Reply:
Already fixed.